Extensions/classic_tags.js

```diff
@@ -174,6 +192,68 @@ XKit.extensions.classic_tags = new Object({
 		}
 	},
 
+	update_tag_counts: function (next_update) {
+		var self = this;
+		var new_post_count_promises = [];
+
+		function fetch_count(tag_name) {
+			var promise = self.get_unread_post_count_for_tag(tag_name);
+			new_post_count_promises.push(promise);
+			return promise;
+		}
+
+		if (self.preferences.show_tags_on_sidebar.value) {
+			var list = $("#xtags");
+			var list_hidden = list.hasClass("hidden");
+			$(".xtag").each(function () {
+				var li = $(this);
+				var anchor = li.find(".result_link");
+				var tag_name = anchor.attr("data-tag-result");
+				if (parseInt(self.tagcounts[tag_name], 10) === self.max_posts_per_tag) {
+					return true;
+				}
+
+				fetch_count(tag_name).then(function (count) {
+					if (!count) { return; }
+					if (list_hidden) {
+						list.removeClass("hidden");
+						list_hidden = false;
+					}
+					
+					li.removeClass("hidden");
+					var existing_count = anchor.find(".count");
+					if (existing_count.length) {
+						existing_count.text(count);
+					} else {
+						anchor.find(".result_title").removeClass("no_count").after("<span class='count'>" + count + "</span>");
+					}
+				});
+			});
+		} else {
+			$("#popover_search .result_link").each(function () {
+				var link = $(this);
+				var tag_name = link.attr("data-tag-result");
+				var count = self.tagcounts[tag_name];
+				if (!count || parseInt(count, 10) < self.max_posts_per_tag) {
+					fetch_count(tag_name);
+				}
+			});
+		}
+
+		var search = $("#search_query");
+		var new_label = " [new]";
+		if (self.preferences.show_new_notification.value && search.attr("placeholder").indexOf(new_label) === -1) {
+			$.when.apply($, new_post_count_promises).then(function () {
+				var any_new_posts = Array.prototype.some.call(arguments, function (count) { return !!count; });
+				if (any_new_posts) {
+					search.attr("placeholder", search.attr("placeholder") + new_label);
```

Comment:
The placeholder should be "Search [new]" instead of "Search Tumblr [new]", otherwise
![image](https://cloud.githubusercontent.com/assets/1437010/10982980/7c37dcda-8410-11e5-80a2-50c3f8ae27ce.png)
happens